redbot/core/core_commands.py

```diff
@@ -1776,8 +1776,13 @@ async def avatar(self, ctx: commands.Context, url: str = None):
                 url = url[1:-1]
 
             async with aiohttp.ClientSession() as session:
-                async with session.get(url) as r:
-                    data = await r.read()
+                try:
+                    async with session.get(url) as r:
+                        data = await r.read()
+                except aiohttp.InvalidURL:
+                    return await ctx.send(_("That URL is invalid."))
+                except aiohttp.ClientConnectionError:
```

Comment:
Request can raise subclasses of any of these [1]: `aiohttp.ClientPayloadError`, `aiohttp.ClientResponseError`, `aiohttp.ClientConnectionError`.

If you want to, you can make more specific error messages for each, but I don't think it matters at all, so just using `aiohttp.ClientError`, which is a superclass of all of the exception mentioned above will work just fine here:
```suggestion
                except aiohttp.ClientError:
```

[1] aiohttp reference - https://docs.aiohttp.org/en/stable/client_reference.html#hierarchy-of-exceptions